packages/fluentui/react-northstar/src/components/Button/Button.tsx

```diff
@@ -100,6 +103,171 @@ export type ButtonStylesProps = Pick<
   hasContent?: boolean;
 };
 
+const useButtonStyles = makeStyles([
+  [
+    null,
```

Comment:
To make the API a bit more intuitive would it make sense to create a `builder` kind of API ? either to build selector-styles pairs (decoupling building/making):

```
import { builder } from 'make-styles';

const rawStyles = builder
  .addDefault({...styles}) // no need to declare null selector
  .add({text: false}, {...styles})
```
or makeStyles could use a builder pattern with a fluent API:

```
const useButtonStyles = makeStyles()
  .addDefault({...styles}) // your null selector
  .add({...styles})
  .make();

const buttonStylesMaker = makeStyles()
  .addDefault({...styles}) // your null selector
  .add({...styles});

const useButtonStyles = buttonStylesMaker.make();
```

Reply:
This would also mean you could have project level style makers that could be reused ?

Reply:
`null` selector probably should become `{}`

Reply:
That's fair, do you think the above pattern is unnecessary in that case ?

Reply:
That's an interesting idea @ling1726 :) In makeStyles I was considering something like extend as well:

Option 1:

```jsx
// Let `extends` property reference another hook that becomes injected:
const useButtonStyles = makeStyles( ... );
const useMyButtonStyles = makeStyles({
  extends: useButtonStyles,
  root: { ... },
  icon: { ... }
});
```

Option 2, let the hook have an extendable method:
```jsx
const useMyButtonStyles = useButtonStyles.extend({
  root: ...,
  icon: ...
});
```

Option 3, pass a hook into makeStyles and have it recognize a style hook and extract the rules:

```jsx
const useMyButtonStyles = makeStyles(useButtonStyles, { ... });
```

Reply:
Realized that option 3 is closest to how StyleX works for extending existing classes:

![image](https://user-images.githubusercontent.com/1110944/100802692-bbb6ae80-33de-11eb-9092-2f67465787a3.png)

Reply:
It was covered in offline why we would prefer to use `className` as a plain property can be passed to any elements. 

---

Another issue is related to overrides and their context, it's a real case in `Avatar` + `Status`.

```jsx
const useStatusStyles = makeStyles();

function Status(props) {
  return (
    <div
      className={useStatusStyles({ status: props.status }, props.className)}
    />
  );
}
```

```jsx
const useAvatarStatusStyles = makeStyles();

function Avatar(props) {
  // ⚠️ See usage
  return <Status className={useAvatarStatusStyles({ square: props.square })} />;
}
```

Avatar can be `square`, but `Status` cannot be. With the current implementation overrides will be resolved in `Avatar` and then passed to `Status`.

In "stylex":

- they should be resolved in `Status` and it means that it somehow should know `square`
- they should be resolved in `Avatar` and it means that it has the same issue as the existing solution in v0 where we have sometimes `className` and `styles`